packages/vue-3/src/EditorContent.ts

```diff
@@ -46,11 +46,10 @@ export const EditorContent = defineComponent({
           if (instance) {
             editor.appContext = {
               ...instance.appContext,
-              provides: Object.assign(
-                instance.appContext.provides,
-                // @ts-ignore
-                instance.provides,
-              ),
+              // Vue internally uses prototype chain to forward/shadow injects across the entire component chain
+              // so don't use object spread operator or 'Object.assign' and just set `provides` as is on editor's appContext
+              // @ts-expect-error forward instance's 'provides' into appContext
+              provides: instance.provides,
             }
```

Comment:
Just to make sure, does this work for both the options API & composition API. I'm not familiar enough with vue to know

Reply:
Yes this will work for both Options and Composition API. Tbh I wrote tests in Composition first and then converted to Options for consistency with other examples. 

Also no need to worry about composition vs options at all: Options API is literally implemented on top of Composition primitives.

Reply:
Thanks for the confirmation!